cmd/traffic/cmd/manager/managerutil/systema.go

```diff
@@ -15,6 +16,10 @@ type SystemaCRUDClient interface {
 	a8rcloud.Closeable
 }
 
+func AgentIsProprietary(ctx context.Context) bool {
+	return strings.Contains(GetAgentImage(ctx), "datawire.io/telepresence-traffic-agent")
```

Comment:
Another comment on this. It's extremely easy for a skilled user to circumvent this and simply return `false` here, recompile the traffic-manager, and suddenly have a system that isn't really restricted.